lib/fathead/python_coverage/parse.py

```diff
@@ -0,0 +1,281 @@
+        documentation of a class
+        :param raw_data: HTML of class scraped from documentation
+        :param base_url: URL of API Documentation
+        """
+        self.raw_data = raw_data
+        self.base_url = base_url
+        self.base_title = base_title
+        self.method_list = []
+        self.class_name = ''
+        self.parse_raw_class_data_into_methods()
+
+    def get_method_list(self):
+        """
+        Used to get list of methods parsed from class
+        :return: List of methods parsed from class
+        """
+        return self.method_list
+
+    def get_method_count(self):
+        """
+        Used to get number of methods
+        :return: number of methods
+        """
+        return len(self.method_list)
+
+    def parse_raw_class_data_into_methods(self):
+        """
+        Scrapes methods from html of class object
+        """
+        soupy = BeautifulSoup(self.raw_data, 'html.parser')
+        self.class_name = soupy.dt["id"]
+        methods = soupy.find_all('dl', {'class': 'method'})
+        for method in set(methods):
+            self.method_list.append(method)
+
+    def get_raw_method(self, method_index):
+        """
+        Returns method from method list at method_index
+        :param method_index: Index of method to return raw data
+        :return: raw html data of method
+        """
+        return self.method_list[method_index]
+
+    def get_parsed_method_from_data(self, method):
+        """
+        Parses method html into json object
+        :param method: raw html data of method
+        :return: json object of parsed values
+        """
+        attributes = {}
+
+        html_soup = BeautifulSoup(str(method), 'html.parser')
+        attributes["base_url"] = self.base_url
+        attributes["base_title"] = self.base_title
+        attributes["id"] = html_soup.dt["id"]
+        attributes["title"] = attributes["id"].split('.')[1] + " " + \
+            attributes["id"].split('.')[2]
+        attributes["headerlink"] = "{}{}".format(self.base_url,
+                                                 html_soup.a["href"])
+
+        attributes["description"] = ''
+        for index, value in enumerate(html_soup.dd.text):
+            attributes["description"] += str(value)
+        attributes["description"] = attributes["description"]\
+            .replace('\n', '\\n').strip('\\n ')
+
+        attributes["code"] = ''
+        # find the line of code in the method
+        # due to formatting, its full of <span> and <em> tags
+        code_line = re.search('^<code.*\n', method, re.MULTILINE).group(0)
+        # grab everything starting with <code to the last </span> tag, which
+        # is the closing paren
+        code_line = code_line[0:code_line.rindex('</span>') + 7]
+        # last step, remove all the html tags to give a clean output
+        # clear_exclude(which='exclude')
+        attributes["code"] = re.sub('<[^>]*>', '', code_line, re.UNICODE)
```

Comment:
Likewise we want to grab the plaintext for their code blocks:

```
<div class="highlight-default"><div class="highlight"><pre>
```

and replace it with our codeblock syntax:

```
<pre><code>
```

This way we'll preserver the look/layout of their docs, and we'll also be able to show syntax highlighted code blocks!